tests/helpers/perf_results_helper.rb

```diff
@@ -757,5 +760,73 @@ def extract_puppetserver_metrics_from_json(file)
 
     row
   end
+
+  # Execute the specified SQL query via BigQuery
+  #
+  # @author Bill Claytor
+  #
+  # @raise [StandardError] If the GOOGLE_APPLICATION_CREDENTIALS environment variable is not set
+  #
+  # @return [Array] The array of versions
+  #
+  # @example
+  #   versions_array = baseline_versions
+  #
+  def query_bigquery(sql)
+    error = "The GOOGLE_APPLICATION_CREDENTIALS environment variable must be set with the location of the JSON key file"
+    raise error if ENV["GOOGLE_APPLICATION_CREDENTIALS"].nil? || ENV["GOOGLE_APPLICATION_CREDENTIALS"].empty?
+
+    bigquery = Google::Cloud::Bigquery.new project: BIGQUERY_PROJECT
+    data = bigquery.query sql
+    logger.error("Cannot find result that matches query: #{sql}") if data.empty?
+    data
+  end
+
+  # Retrieve the list of baseline PE versions from BigQuery
+  #
+  # @author Bill Claytor
+  #
+  # @raise [StandardError] If the query returns no data
+  #
+  # @return [Array] The array of versions
+  #
+  # @example
+  #   versions_array = baseline_versions
+  #
+  def baseline_versions
+    sql = "SELECT DISTINCT pe_build_number FROM `perf-metrics.perf_metrics.atop_metrics` " \
+          "WHERE pe_build_number IS NOT NULL " \
+          "ORDER BY pe_build_number"
+    data = query_bigquery(sql)
+    raise "Error: cannot find result that matches query: #{sql}" if data.empty?
+
+    versions = []
+    data.each do |row|
+      versions << row[:pe_build_number]
+    end
+
+    versions
+  end
+
+  # Verify that the specified baseline PE version exists in BigQuery
+  #
+  # @author Bill Claytor
+  #
+  # @param [String] baseline_pe_ver The baseline PE version to verify
+  #
+  # @raise [StandardError] If the specified baseline PE version is not found
```

Comment:
I feel like this shouldn't raise an error.  It should be called valid_baseline_version?  and return true or false.  The caller should decide what to do with the information.  Generally the caller will have better information to include in the error as well... like the source of the version.  So it can give a better error.

Reply:
In this case I'm naming it 'verify...' and raising the error here so that it can be easily tested, and if the caller wants to trap it they can do that. However, at this point there is only one caller, which is the `validate:baseline_pe_version` rake task, which we don't currently have a good way to test. I could add a `valid_baseline_version?` method and have this call that, but I was holding off on that until there was a need to check the baseline version without raising an error.

Reply:
Since the caller in a rake task I think it is fine this way